File: hw/pasquel/u3/HW-Strategy/SortingApp/src/main/java/ec/edu/espe/sortingapp/model/BubbleSort.java

package ec.edu.espe.sortingapp.model;

/**
 *
 * @author Marlon Pasquel
 */

public class BubbleSort implements SortingStrategy {
    @Override
    public void sort() {
        System.out.println("Sorting using Bubble Sort");
        int[] array = SortingStrategy.ARRAY.clone(); 
        int n = array.length;
        for (int i = 0; i < n-1; i++) {
            for (int j = 0; j < n-i-1; j++) {
                if (array[j] > array[j+1]) {
                    int temp = array[j];
                    array[j] = array[j+1];
                    array[j+1] = temp;
                }
            }
        }
        System.out.println("Sorted array: " + java.util.Arrays.toString(array));
    }
}

File: EXAM/espinosa/u3/EspinosaStrategy/controller/Bubble_sort.py
import Sorting_strategy

class Bubble_sort(Sorting_strategy):

    def __init__ (self):
        self
        
    
    def Sort(mylist):
 
        for i in range(len(mylist)):
            for j in range(len(mylist)-1):
                if mylist[j+1] < mylist[j]:
                    mylist[j],mylist[j+1] = mylist[j+1],mylist[j]
        return mylist


